pkg/cmd/gitprovider/add.go

```diff
@@ -29,6 +29,8 @@ var GitProviderAddCmd = &cobra.Command{
 		setGitProviderConfig := apiclient.SetGitProviderConfig{}
 		setGitProviderConfig.BaseApiUrl = new(string)
 		setGitProviderConfig.Username = new(string)
+		setGitProviderConfig.SigningKey = new(string)
+		setGitProviderConfig.SigningMethod = new(string)
 
 		gitprovider_view.GitProviderSelectionView(&setGitProviderConfig, nil, false)
```

Comment:
The provider does not seem to return the signing properties from the API.

I added Github and a GPG key but when I do `dtn gp ls -f json` both the fields are empty. Can you confirm that it's working for you?

P.S. The GPG key was correctly set in the git config which means it was stored in the DB but got lost somewhere along the way to the TUI 😄

Reply:
digging into it

Reply:
The value of the fiiels are indead present here in the api but as it is difficult for the api to completly parse them. please take a look, here I tried explicity setting the field in here 
```supportedProviders := config.GetSupportedGitProviders()
		var gitProviderViewList []gitprovider_view.GitProviderView

		for _, gitProvider := range gitProviders {
			for _, supportedProvider := range supportedProviders {
				if gitProvider.Id == supportedProvider.Id {
					gitProviderViewList = append(gitProviderViewList,
						gitprovider_view.GitProviderView{
							Id:       gitProvider.Id,
							Name:     supportedProvider.Name,
							Username: gitProvider.Username,
							SigningKey: *gitProvider.SigningKey,
						},
					)
				}
			}
		}
```  
And I go the following output 
```
[
  {
    "Id": "github",
    "Name": "GitHub",
    "Username": "bryans-go",
    "BaseApiUrl": "",
    "Token": "",
    "SigningMethod": "",
    "SigningKey": "ssh-ed25519 abc<full key> sabrinabryan1990@gmail.com"
  }
]
```

Reply:
I got it as these things (fields) are optional, which is set by us. thus the API ignores them and only return required fields

Reply:
```
// GitProvider struct for GitProvider
type GitProvider struct {
	BaseApiUrl    *string        `json:"baseApiUrl,omitempty"`
	Id            string         `json:"id"`
	SigningKey    *string        `json:"signingKey,omitempty"`
	SigningMethod *SigningMethod `json:"signingMethod,omitempty"`
	Token         string         `json:"token"`
	Username      string         `json:"username"`
}
```

Reply:
There is nothing we can do in it

Reply:
> The provider does not seem to return the signing properties from the API.
> 
> I added Github and a GPG key but when I do `dtn gp ls -f json` both the fields are empty. Can you confirm that it's working for you?
> 
> P.S. The GPG key was correctly set in the git config which means it was stored in the DB but got lost somewhere along the way to the TUI 😄

@Tpuljak The same thing happens with the base-api url . see here after adding base-api url when adding gitea as git provider it shows empty field for  "BaseApiUrl": "",
```
  {
    "Id": "gitea",
    "Name": "Gitea",
    "Username": "bryans",
    "BaseApiUrl": "",
    "Token": "",
    "SigningMethod": "",
    "SigningKey": ""
  }
]
```

Reply:
@bryans-go seems you misunderstood me a bit. I don't have an issue with empty fields being returned if they truly are empty in the DB. The issue I had was that the fields were empty even though I set them in the DB.

Reply:
Let me retest with the latest changes

Reply:
Ah, you were on the right track.

https://github.com/bryans-go/daytona/blob/commits-git/pkg/cmd/gitprovider/list.go#L48

The issue is not the response from the API, the issue is that the fields were not mapped before outputting. You need to add the signing method in the loop above. No need to add the signing key since it will be removed on the controller with https://github.com/daytonaio/daytona/pull/1146#discussion_r1781143422

Reply:
This doesn't seem to be addressed again

Reply:
done